From b9fc66d2feb721d005bdcf5f7ca23fdae9f3bca7 Mon Sep 17 00:00:00 2001
From: Wei-Luan Wang <weiluanwang@google.com>
Date: Thu, 26 Jun 2025 03:05:20 +0000
Subject: [PATCH] disable boringssl in base::RandBytes

Introduced in r1045239 (crrev.com/c/3880863).
Updated in r1321254 (crrev.com/c/5669584).
Updated in r1477790 (crrev.com/c/6665689).
Updated in r1624070 (crrev.com/c/7812144).

Change-Id: I2cdc2fe34a83b8debad12c0143e881a7a52c5bf5
---
 base/rand_util_posix.cc | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/base/rand_util_posix.cc b/base/rand_util_posix.cc
index 260d30c873..4bcbc75f6f 100644
--- a/base/rand_util_posix.cc
+++ b/base/rand_util_posix.cc
@@ -22,7 +22,6 @@
 #include "base/system/sys_info.h"
 #include "base/time/time.h"
 #include "build/build_config.h"
-#include "third_party/boringssl/src/include/openssl/rand.h"
 
 #if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
 #include "third_party/lss/linux_syscall_support.h"
@@ -92,12 +91,6 @@ bool GetRandomSyscall(void* output, size_t output_length) {
 namespace {
 
 void RandBytesInternal(span<uint8_t> output, bool avoid_allocation) {
-  // The BoringSSL experiment takes priority over everything else.
-  if (!avoid_allocation) {
-    // BoringSSL's RAND_bytes always returns 1. Any error aborts the program.
-    (void)RAND_bytes(output.data(), output.size());
-    return;
-  }
 #if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_ANDROID)
   // On Android it is mandatory to check that the kernel _version_ has the
   // support for a syscall before calling. The same check is made on Linux and
-- 
2.54.0.545.g6539524ca2-goog

